### PR TITLE
check pid in zip

### PR DESCRIPTION
**Pull request by shu65**
This PR solves `ZipContainer` issue in https://github.com/chainer/chainerio/issues/104

### CODE REVIEW

File: chainerio/containers/zip.py [hunk 1]
```diff
@@ -29,9 +32,15 @@ def _check_zip_file_name(self, base):
         filename, file_extension = os.path.splitext(self.base)
 
     def _open_zip_file(self, mode='r'):
-        mode = mode.replace("b", "")
+        if self.zip_file_obj is not None:
+            if self.zip_file_obj_pid != os.getpid():
+                self.zip_file_obj = None
+                self.zip_file_obj_pid = None
+                self.zip_file_obj_mode = None
```
Comment (belldandyxtq) [inline chainerio/containers/zip.py]: I do not think you need to save the "mode"

File: tests/container_tests/test_zip_container.py [hunk 2]
```diff
@@ -427,3 +433,71 @@ def test_stat(self):
...
+
+            p1 = multiprocessing.Process(target=func)
+            p2 = multiprocessing.Process(target=func)
+            p1.start()
+            p2.start()
+
+            p1.join()
+            p2.join()
+
+            self.assertEqual(p1.exitcode, 0)
```
Comment (belldandyxtq) [inline tests/container_tests/test_zip_container.py]: Can you please add a timeout mechanism as the problem here is hanging?

File: chainerio/containers/zip.py (hunk 1 above)
Comment (shu65) [reply chainerio/containers/zip.py]: `mode` requires to check the mode of previous zip.

File: chainerio/containers/zip.py [hunk 3]
```diff
@@ -30,6 +32,15 @@ def _check_zip_file_name(self, base):
...
+            self._close_zip_file()
+
+        if self.zip_file_obj is not None \
+                and self.zip_file_obj_pid != os.getpid():
+            self.zip_file_obj = None
+            self.zip_file_obj_pid = None
+            self.zip_file_obj_mode = None
 
         if self.zip_file_obj is None:
             zip_file = self.base_handler.open(self.base, "rb")
```
Comment (belldandyxtq) [inline chainerio/containers/zip.py]: Can you please match the mode here as well?

File: chainerio/containers/zip.py [hunk 4]
```diff
@@ -80,12 +94,14 @@ def _wrap_fileobject(self, file_obj: Type['IOBase'],
     def open(self, file_path, mode='r',
              buffering=-1, encoding=None, errors=None,
              newline=None, closefd=True, opener=None):
+        if sys.version_info < (3, 6) and "w" in mode:
+            raise ValueError('Mode w and wb is not supported')
```
Comment (belldandyxtq) [inline chainerio/containers/zip.py]: Can you mention that "w" and "wb" is not supported only in Python < 3.6?

**Pull request merged**